### PR TITLE
test: add benchmark GitHub workflow

### PR DESCRIPTION
**Pull request by yonadaaa**
Adds a GitHub Action where we can run benchmarks on a remote machine and inspect the output.

### CODE REVIEW

**Comment by yonadaaa** [comment]
> Is there a way for us to configure the benchmark suite to report absolute numbers instead of relative ones? ("x seconds" vs "x times faster than y")

Yes, it logs a full table of stats before "x times faster than y". I'll look into what the numbers mean and see if we can compress them down to something more legible: 
image

**Comment by yonadaaa** [comment]
> Also curious if you looked into actions like this one https://github.com/benchmark-action/github-action-benchmark that have built-in support for reporting the benchmark results to a github-pages branch

My impression is that most Actions only support certain benchmark tools (ie. not Vitest/tinybench). However [github-action-benchmark](https://github.com/benchmark-action/github-action-benchmark) does mention some custom `biggerIsBetter` that we can use as an adapter - let me try that too :)

**Comment by alvrs** [comment]
> Yes, it logs a full table of stats before "x times faster than y". I'll look into what the numbers mean and see if we can compress them down to something more legible:

Ahh that's great. Somehow didn't find it in the action logs

**Comment by yonadaaa** [comment]
> Ahh that's great. Somehow didn't find it in the action logs 

I had `--silent` on some of the runs which I think only prints the "x > y" summary.

File: .github/workflows/bench.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Benchmarks
+
+on:
+  workflow_dispatch:
+
+jobs:
+  test:
```
Comment (holic) [inline .github/workflows/bench.yml:7]:
```suggestion
  benchmarks:
```

**Review by holic — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
   "scripts": {
     "all-codegen": "for dir in packages/store packages/world packages/world-modules packages/cli e2e/packages/contracts examples/*/packages/contracts templates/*/packages/contracts; do (cd \"$dir\" && pwd && pnpm build); done",
     "all-install": "for dir in . docs e2e examples/* templates/*; do (cd \"$dir\" && pwd && pnpm install); done",
+    "bench": "pnpm run --filter=store-sync bench",
```
[the remark this answers is not in the document]
Comment (holic) [reply package.json:13]: I think if we do `pnpm run --recursive bench`, it'll only run packages in the workspace with the bench script

Comment (holic) [inline package.json:13]:
```suggestion
    "bench": "pnpm run --recursive bench",
```

**Review by holic — APPROVED** [review]

**Pull request merged**